src/firefly/js/visualize/ui/FileUploadViewPanel.jsx

```diff
@@ -72,46 +57,57 @@ const TABLES_ONLY_SUPPORTED_TYPES=[
     FileAnalysisType.Table,
 ];
 
-
-const isTablesOnly= () => getAppOptions()?.uploadPanelLimit==='tablesOnly';
-
-
 const uploadOptions = 'uploadOptions';
 
-let currentAnalysisResult, currentReport, currentSummaryModel, currentDetailsModel;
 const FILE_UPLOAD_KEY= 'file-upload-key-';
 let keyCnt=0;
 
-export function FileUploadViewPanel({setSubmitText}) {
+export function FileUploadViewPanel({setSubmitText, acceptMoc}) {
+
+    const {groupKey}= useContext(FieldGroupCtx);
 
-    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(panelKey, {isLoading:false,statusKey:''}));
+    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(groupKey, {isLoading:false,statusKey:''}));
     const isWsUpdating          = useStoreConnector(() => isAccessWorkspace());
-    const uploadSrc             = useStoreConnector(() => getFieldVal(panelKey, uploadOptions));
-    const {message, analysisResult, report, summaryModel, detailsModel} = useStoreConnector((oldState) => getNextState(oldState));
+    const uploadSrc             = useStoreConnector(() => getFieldVal(groupKey, uploadOptions));
+
+
+    const SUMMARY_TBL_ID = groupKey; //FileUploadAnalysis
+    const DETAILS_TBL_ID = groupKey + '-Details';
+    const UNKNOWN_FORMAT = 'UNKNOWN';
+    const summaryUiId = SUMMARY_TBL_ID + '-UI';
+    const detailsUiId = DETAILS_TBL_ID + '-UI';
+
+    let {message, analysisResult, report, summaryModel, detailsModel, prevAnalysisResult} = useStoreConnector((oldState) =>
+        getNextState(oldState, groupKey));
 
     const [loadingMsg,setLoadingMsg]= useState(() => '');
     const [uploadKey,setUploadKey]= useState(() => FILE_UPLOAD_KEY+keyCnt);
 
     useEffect(() => {
-        setSubmitText(getLoadButtonText());
-    },[report,setSubmitText, summaryModel, detailsModel]);
+        dispatchValueChange({fieldKey:getLoadingFieldName(groupKey), groupKey:groupKey, prevAnalysisResult: analysisResult});
+    }, [analysisResult]);
 
     useEffect(() => {
-        if (message || (analysisResult && analysisResult !== currentAnalysisResult)) {
-            if (currentAnalysisResult !== analysisResult) {
-                currentAnalysisResult = analysisResult;
-            }
-            dispatchComponentStateChange(panelKey, {isLoading: false});
-        }
-    });
+        dispatchValueChange({fieldKey:getLoadingFieldName(groupKey), groupKey:groupKey, report: report});
+    }, [report]);
+
+    useEffect(() => {
+        if (message || analysisResult) dispatchComponentStateChange(groupKey, {isLoading: false});
+    }, [message, analysisResult]);
+
```

Comment:
This might be working but it is not really what it is intend for. This is where you should use `useFieldGroupMetaState`

Something like.....

```js
const [getUploadMetaInfo, setUploadMetaInfo]= useFieldGroupMetaState({summaryModel:undefined, analysisResult: undefined, etc etc })
```

```js
useEffect(() => {
    setUploadMetaInfo({getUploadMetaInfo(), analysisResult,detailsModel, etc })
}[analysisResult,detailsModel, etc ])
```